src/games/strategy/engine/data/Unit.java

```diff
@@ -12,16 +15,30 @@
   private int m_hits = 0;
   private final UnitType m_type;
 
+
+  public static Unit createUnit(final UnitType unitType) {
+    return createUnit(unitType, null);
+  }
+
+  public static Unit createUnit(final UnitType unitType, final PlayerID player) {
+    return unitType.create(player);
+  }
+
+  public static Unit createUnit(final String typeName, final GameData data) {
+    return createUnit(typeName, data, null);
+  }
+
+  public static Unit createUnit(final String typeName, final GameData data, final PlayerID owner) {
+    return createUnit( (new UnitType(typeName, data)), owner);
+  }
+
```

Comment:
The DefaultUnitFactory is overkill for what it is doing. I think the intent was to have the createUnit methods as a starting point for some further refactor work. I suspect I broke off some additional commits from this PR to form a new branch, and these changes were leftover in this branch. Regardless, have removed them.